test/boost/auth_resource_test.cc

```diff
@@ -23,6 +22,11 @@ std::ostream& operator<<(std::ostream& os, resource_kind kind) {
     return os;
 }
 
+std::ostream& operator<<(std::ostream& os, const std::unordered_set<resource>& resources) {
```

Comment:
i'd suggest s/std::unordered_set<resource>/resource_set/, so it's more clear what we intend to format.

Reply:
also, just a nit, might want to define `boost_test_print_type()` instead, so it's more obvious what this function is for.

Reply:
Sure, fixed - it looks better now. Thanks!

Also, I replaced the `operator<<(std::ostream& os, resource_kind kind)` above with `boost_test_print_type()` to keep consistency in this file :+1: